src/main/java/com/pgs/booking/service/UserService.java

```diff
@@ -13,15 +19,53 @@
 import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
 import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class UserService implements UserDetailsService, AuthenticationUserDetailsService<PreAuthenticatedAuthenticationToken> {
 
     private final UserRepository userRepository;
+    private final UserDtoMapper userDtoMapper;
+    private final CreateUserDtoMapper createUserDtoMapper;
+    private final RoleRepository roleRepository;
     private final TokenStore tokenStore;
 
+    public UserDto getSingleUser(Long id) {
+        var user = getUserById(id);
+        return userDtoMapper.mapToUserDto(user);
+    }
+
+    public UserDto addUser(CreateUserDto createUserDto) {
+        var savedUser = userRepository.save(createUserDtoMapper.mapToUser(createUserDto));
+        return userDtoMapper.mapToUserDto(savedUser);
+    }
+
+    public UserDto activateUser(Long id) {
+        var user = getUserById(id);
+        user.setEnabled(true);
+        var savedUser = userRepository.save(user);
+        return userDtoMapper.mapToUserDto(savedUser);
+    }
+
+    public UserDto deactivateUser(Long id) {
+        var user = getUserById(id);
```

Comment:
`activateUser(Long id)` and `deactivateUser(Long id)` are very similar. Extract a private method with common logic.